test/features/composer/presentation/composer_controller_test.dart

```diff
@@ -0,0 +1,852 @@
+    Get.put<ImagePaths>(ImagePaths());
+    Get.put<ResponsiveUtils>(ResponsiveUtils());
+    Get.put<Uuid>(mockUuid);
+    Get.put<ApplicationManager>(mockApplicationManager);
+    Get.put<ToastManager>(mockToastManager);
+
+    // Mock Getx controllers
+    Get.put<MailboxDashBoardController>(mockMailboxDashBoardController);
+    Get.put<NetworkConnectionController>(mockNetworkConnectionController);
+    Get.put<BeforeReconnectManager>(mockBeforeReconnectManager);
+    Get.put<RichTextMobileTabletController>(mockRichTextMobileTabletController);
+
+    // Mock composer controller
+    mockLocalFilePickerInteractor = MockLocalFilePickerInteractor();
+    mockLocalImagePickerInteractor = MockLocalImagePickerInteractor();
+    mockGetEmailContentInteractor = MockGetEmailContentInteractor();
+    mockGetAllIdentitiesInteractor = MockGetAllIdentitiesInteractor();
+    mockUploadController = MockUploadController();
+    mockRemoveComposerCacheOnWebInteractor = MockRemoveComposerCacheOnWebInteractor();
+    mockSaveComposerCacheOnWebInteractor = MockSaveComposerCacheOnWebInteractor();
+    mockDownloadImageAsBase64Interactor = MockDownloadImageAsBase64Interactor();
+    mockTransformHtmlEmailContentInteractor = MockTransformHtmlEmailContentInteractor();
+    mockGetAlwaysReadReceiptSettingInteractor = MockGetAlwaysReadReceiptSettingInteractor();
+    mockCreateNewAndSendEmailInteractor = MockCreateNewAndSendEmailInteractor();
+    mockCreateNewAndSaveEmailToDraftsInteractor = MockCreateNewAndSaveEmailToDraftsInteractor();
+    
+    composerController = ComposerController(
+      mockLocalFilePickerInteractor,
+      mockLocalImagePickerInteractor,
+      mockGetEmailContentInteractor,
+      mockGetAllIdentitiesInteractor,
+      mockUploadController,
+      mockRemoveComposerCacheOnWebInteractor,
+      mockSaveComposerCacheOnWebInteractor,
+      mockDownloadImageAsBase64Interactor,
+      mockTransformHtmlEmailContentInteractor,
+      mockGetAlwaysReadReceiptSettingInteractor,
+      mockCreateNewAndSendEmailInteractor,
+      mockCreateNewAndSaveEmailToDraftsInteractor
+    );
+  });
+
+  tearDown(() {
+    Get.reset();
+    composerController = null;
+  });
+  
+  group('ComposerController test:', () {
+    group('hash draft email test:', () {
+      void grandPermission(Permission permission) {
+        TestDefaultBinaryMessengerBinding
+          .instance
+          .defaultBinaryMessenger
+          .setMockMethodCallHandler(
+            const MethodChannel('flutter.baseflow.com/permissions/methods'), 
+            (MethodCall methodCall) async {
+              if (methodCall.arguments == permission.value) {
+                return PermissionStatus.granted.index;
+              }
+
+              return PermissionStatus.denied.index;
+            }
+          );
+      }
+
+      test(
+        'should update _savedEmailDraftHash '
+        'when there is a new view state '
+        'and the state is GetAlwaysReadReceiptSettingSuccess',
```

Comment:
What is the point of this unit test? Why update _savedEmailDraftHash when state is GetAlwaysReadReceiptSettingSuccess

Reply:
read receipt is one of the value in the hash, and its value is retrieved from the server. Hence, `GetAlwaysReadReceiptSettingSuccess` is needed.